// angular-google-charts/esm2022/lib/standalone.mjs
// Standalone component versions
import { Component } from '@angular/core';
import { GoogleChartComponent } from './components/google-chart/google-chart.component';
import { ChartWrapperComponent } from './components/chart-wrapper/chart-wrapper.component';
import { DashboardComponent } from './components/dashboard/dashboard.component';
import { ControlWrapperComponent } from './components/control-wrapper/control-wrapper.component';
import { ChartEditorComponent } from './components/chart-editor/chart-editor.component';
import * as i0 from "@angular/core";
// Create standalone versions by copying the metadata and setting standalone: true
export class GoogleChart extends GoogleChartComponent {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: GoogleChart, deps: null, target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: GoogleChart, isStandalone: true, selector: "google-chart", host: { classAttribute: "google-chart" }, exportAs: ["googleChart"], usesInheritance: true, ngImport: i0, template: '', isInline: true, styles: [":host{width:-moz-fit-content;width:fit-content;display:block}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: GoogleChart, decorators: [{
            type: Component,
            args: [{ selector: 'google-chart', template: '', host: { class: 'google-chart' }, exportAs: 'googleChart', standalone: true, styles: [":host{width:-moz-fit-content;width:fit-content;display:block}\n"] }]
        }] });
export class ChartWrapper extends ChartWrapperComponent {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: ChartWrapper, deps: null, target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: ChartWrapper, isStandalone: true, selector: "chart-wrapper", host: { classAttribute: "chart-wrapper" }, exportAs: ["chartWrapper"], usesInheritance: true, ngImport: i0, template: '', isInline: true, styles: [":host{width:-moz-fit-content;width:fit-content;display:block}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: ChartWrapper, decorators: [{
            type: Component,
            args: [{ selector: 'chart-wrapper', template: '', host: { class: 'chart-wrapper' }, exportAs: 'chartWrapper', standalone: true, styles: [":host{width:-moz-fit-content;width:fit-content;display:block}\n"] }]
        }] });
export class Dashboard extends DashboardComponent {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: Dashboard, deps: null, target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: Dashboard, isStandalone: true, selector: "dashboard", host: { classAttribute: "dashboard" }, exportAs: ["dashboard"], usesInheritance: true, ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: Dashboard, decorators: [{
            type: Component,
            args: [{
                    selector: 'dashboard',
                    template: '<ng-content></ng-content>',
                    exportAs: 'dashboard',
                    host: { class: 'dashboard' },
                    standalone: true
                }]
        }] });
export class ControlWrapper extends ControlWrapperComponent {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: ControlWrapper, deps: null, target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: ControlWrapper, isStandalone: true, selector: "control-wrapper", host: { classAttribute: "control-wrapper" }, exportAs: ["controlWrapper"], usesInheritance: true, ngImport: i0, template: '', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: ControlWrapper, decorators: [{
            type: Component,
            args: [{
                    selector: 'control-wrapper',
                    template: '',
                    host: { class: 'control-wrapper' },
                    exportAs: 'controlWrapper',
                    standalone: true
                }]
        }] });
export class ChartEditor extends ChartEditorComponent {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: ChartEditor, deps: null, target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: ChartEditor, isStandalone: true, selector: "chart-editor", host: { classAttribute: "chart-editor" }, usesInheritance: true, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: ChartEditor, decorators: [{
            type: Component,
            args: [{
                    selector: 'chart-editor',
                    template: `<ng-content></ng-content>`,
                    host: { class: 'chart-editor' },
                    standalone: true
                }]
        }] });
// Re-export provider function
export { provideGoogleCharts } from './providers';
